crates/core/src/sql/value/value.rs

```diff
@@ -477,6 +477,49 @@ impl Value {
 	// Simple output of value type
 	// -----------------------------------
 
+	pub fn kind(&self) -> Option<Kind> {
+		match self {
+			Value::None => None,
+			Value::Null => Some(Kind::Null),
+			Value::Bool(_) => Some(Kind::Bool),
+			Value::Number(_) => Some(Kind::Number),
+			Value::Strand(_) => Some(Kind::String),
+			Value::Duration(_) => Some(Kind::Duration),
+			Value::Datetime(_) => Some(Kind::Datetime),
+			Value::Uuid(_) => Some(Kind::Uuid),
+			Value::Array(arr) => Some(Kind::Array(
+				Box::new(arr.first().and_then(|v| v.kind()).unwrap_or_default()),
+				None,
+			)),
+			Value::Object(_) => Some(Kind::Object),
+			Value::Geometry(geo) => Some(Kind::Geometry(vec![geo.as_type().to_string()])),
+			Value::Bytes(_) => Some(Kind::Bytes),
+			Value::Thing(_) => None, // TODO: Should this have a kind?
+			Value::Param(_) => None,
+			Value::Idiom(_) => None,
+			Value::Table(_) => None,
+			Value::Mock(_) => None,
+			Value::Regex(_) => None,
+			Value::Cast(_) => None,
+			Value::Block(_) => None,
+			Value::Range(_) => None,
+			Value::Edges(_) => None,
+			Value::Future(_) => None,
+			Value::Constant(_) => None,
+			Value::Function(_) => Some(Kind::Function(None, None)),
```

Comment:
This is a function invocation, not a function value. I believe the closure value instead should be the function type